sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionFacadeEjb.java

```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2018 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *******************************************************************************/
+package de.symeda.sormas.backend.action;
+
+import java.sql.Timestamp;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.ejb.EJB;
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import de.symeda.sormas.api.action.ActionCriteria;
+import de.symeda.sormas.api.action.ActionDto;
+import de.symeda.sormas.api.action.ActionFacade;
+import de.symeda.sormas.api.action.ActionStatEntry;
+import de.symeda.sormas.api.user.UserRole;
+import de.symeda.sormas.backend.event.EventFacadeEjb;
+import de.symeda.sormas.backend.event.EventService;
+import de.symeda.sormas.backend.user.User;
+import de.symeda.sormas.backend.user.UserFacadeEjb;
+import de.symeda.sormas.backend.user.UserService;
+import de.symeda.sormas.backend.util.DtoHelper;
+import de.symeda.sormas.backend.util.ModelConstants;
+
+@Stateless(name = "ActionFacade")
+public class ActionFacadeEjb implements ActionFacade {
+
+	@PersistenceContext(unitName = ModelConstants.PERSISTENCE_UNIT_NAME)
+	private EntityManager em;
+
+	@EJB
+	private ActionService actionService;
+	@EJB
+	private UserService userService;
+	@EJB
+	private EventService eventService;
+
+	public Action fromDto(ActionDto source) {
+
+		if (source == null) {
+			return null;
+		}
+
+		boolean creation = false;
+		Action target = actionService.getByUuid(source.getUuid());
+		if (target == null) {
+			creation = true;
+			target = new Action();
+			target.setUuid(source.getUuid());
+			if (source.getCreationDate() != null) {
+				target.setCreationDate(new Timestamp(source.getCreationDate().getTime()));
+			}
+		} else {
+			target.setReplyingUser(userService.getCurrentUser());
```

Comment:
If Event Actions should ever be part of the mobile app, this will lead to problems. You should instead set the replying user in the UI or wherever the action is built and add `target.setReplyingUser(userService.getByReferenceDto(source.getReplyingUser())` below because otherwise, this data will always be lost when the action has been edited in the mobile app

Reply:
OK